web/src/ui_init.js

```diff
@@ -435,6 +436,15 @@ export function initialize_everything() {
               `stream_data` in `page_params` itself
     */
 
+    const ls = localstorage();
+    const untrackedFlaggedMessages = ls.get("untrackedFlaggedMessages");
+    if (untrackedFlaggedMessages) {
+        for (const message of untrackedFlaggedMessages) {
+            message_flags.send_flag_update_for_messages(message.msg_ids, message.flag, message.op);
+        }
+        ls.remove("untrackedFlaggedMessages");
+    }
```

Comment:
Please don't put new code inline in `ui_init.js`; that module should call a function in the `message_flags` module that contains this logic, to have reasonable abstractions.

Also, we likely certainly shouldn't do this as the very first thing before other initialization is completed.